### PR TITLE
Improve FindCACTUS

### PR DESCRIPTION
**Pull request by lmoureaux**
Bring it to the same level as FindxDAQ. Most of the code was copied from the latter, with support for threading libraries removed.

### CODE REVIEW

File: FindCACTUS.cmake [hunk 1]
```diff
@@ -1,145 +1,232 @@
...
-endfunction()
+endmacro()
 
 # List all supported libs and their dependencies
-cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
-cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
-cactus_library(amc13_tools HEADER "amc13/AMC13.hh" DEPENDS amc13_amc13)
+_cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
+_cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
+_cactus_library(amc13_tools HEADER "amc13/AMC13.hh" DEPENDS amc13_amc13)
```
Comment (lpetre-ulb) [inline FindCACTUS.cmake]: Are you sure that `amc13/AMC13.hh` is the header file for `amc13_tools` ? The source code and the RPM do not install any header file for the AMC13 tools library. Is your code able to handle such case ?

File: FindCACTUS.cmake [hunk 2]
```diff
@@ -1,145 +1,232 @@
...
-    set(cactus_all_libs "${cactus_all_libs}" PARENT_SCOPE)
-endfunction()
+endmacro()
 
 # List all supported libs and their dependencies
-cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
-cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
-cactus_library(amc13_tools HEADER "amc13/AMC13.hh" DEPENDS amc13_amc13)
+_cactus_library(uhal_uhal HEADER "uhal/uhal.hpp")
+_cactus_library(amc13_amc13 HEADER "amc13/AMC13Simple.hh")
```
Comment (lpetre-ulb) [inline FindCACTUS.cmake]: The two AMC13 libraries do not have their SONAME set, so I think the option should be added.

File: FindCACTUS.cmake [hunk 3]
```diff
@@ -1,145 +1,232 @@
...
+list(REMOVE_DUPLICATES CACTUS_LIBRARIES)
+
+find_package_handle_standard_args(
+    CACTUS
+    FOUND_VAR CACTUS_FOUND
+    REQUIRED_VARS CACTUS_INCLUDE_DIRS CACTUS_LIBRARIES
+    VERSION_VAR CACTUS_VERSION
+    HANDLE_COMPONENTS)
+
+# i2o requires an additional definition
```
Comment (lpetre-ulb) [inline FindCACTUS.cmake]: Does CACTUS need and/or provide `i2o` ? Aren't these line a leftover from `xDAQ` ?

File: FindCACTUS.cmake [hunk 4]
```diff
@@ -1,145 +1,232 @@
...
 
 # Declares a library that can be required using the COMPONENTS argument
-function(cactus_library name)
-    cmake_parse_arguments(ARG "THREADS" "HEADER" "DEPENDS" "" ${ARGN})
+macro(_cactus_library name)
+    cmake_parse_arguments(ARG "NO_SONAME" "HEADER" "DEPENDS" ${ARGN})
 
-    set(cactus_${name}_header ${ARG_HEADER} PARENT_SCOPE)
-    set(cactus_${name}_depends ${ARG_DEPENDS} PARENT_SCOPE)
+    set(cactus_${name}_depends ${ARG_DEPENDS})
```
Comment (lpetre-ulb) [inline FindCACTUS.cmake:50]: Don't you also need to set `cactus_${name}_header` for the header files ?

**Review by lpetre-ulb — CHANGES_REQUESTED** [review]
I think small adjustments are needed. If I'm mistaken, I'll be happy to approve the PR.

File: FindCACTUS.cmake (hunk 1 above)
Comment (lmoureaux) [reply FindCACTUS.cmake]: How would one compile against a library without the corresponding headers? But `AMC13.hh` is indeed wrong.

Comment (lpetre-ulb) [reply FindCACTUS.cmake]: I have absolutely no idea... Can't `amc13_tools` be an internal library which is not supposed to be linked to by external binaries ? In the GEMs, I have already seen libraries whose headers must be copied in every source tree... :-/

Comment (lmoureaux) [reply FindCACTUS.cmake]: `amc13_tools` is not required by `cmsgemos`.

**Comment by lmoureaux** [comment]
Rebased and addressed your remarks.

**Review by lpetre-ulb — APPROVED** [review]

**Comment by lpetre-ulb** [comment]
> Rebased and addressed your remarks.

And more. ;-) Thanks !

**Pull request merged**